pertpy/tools/_perturbation_space/_discriminator_classifier.py

```diff
@@ -67,9 +86,34 @@ def load(  # type: ignore
         if layer_key is not None and layer_key not in adata.obs.columns:
             raise ValueError(f"Layer key {layer_key} not found in adata.")
 
+        if embedding_key is not None and embedding_key not in adata.obsm.keys():
+            raise ValueError(f"Embedding key {embedding_key} not found in adata.obsm.")
```

Comment:
True, we don't really need it. However, we do throw a separate error for most other perturbation spaces with the `embedding_key` as a parameter, so that's why for consistency, I also implemented it here.